src/agent/Core/Controller/CheckoutSession.cpp

```diff
@@ -178,7 +179,21 @@ Controller::initiateSession(Client *client, Request *req) {
 	TRACE_POINT();
 	req->sessionCheckoutTry++;
 	try {
-		req->session->initiate(false);
+		if (!req->session->initiate()) {
+			unsigned long long timeout = 100000;
+			while (timeout > 0 && !waitUntilWritable(req->session->fd(), &timeout)){}
```

Comment:
This is a blocking wait. In the Controller you're supposed to use a libev io watcher to use the event loop to wait.